### PR TITLE
Moved samplers module into pulse.pulse_lib.samplers

### LINKED ISSUE #2868: Move pulse samplers into pulse_lib module
### What is the expected enhancement?
Samplers are only used for constructing pulses. These should be moved within the pulse_lib module.

I believe these are internal only for now and should be movable without using a shim. Thoughts @mtreinish?

### PR DESCRIPTION
**Pull request by taalexander**
### Summary
Closes #2868. This module moves pulse.samplers into pulse.pulse_lib.samplers.

### CODE REVIEW

File: qiskit/pulse/pulse_lib/samplers/decorators.py [hunk 1]
```diff
@@ -135,9 +135,10 @@ def linear(times: np.ndarray, m: float, b: float) -> np.ndarray:
 
 import numpy as np
 
-from qiskit.pulse.samplers import strategies
 import qiskit.pulse.commands as commands
 
+from . import strategies
```
Comment (lcapelluto) [inline qiskit/pulse/pulse_lib/samplers/decorators.py]: can we use `from qiskit.pulse.pulse_lib.import` 😍 ? It's okay if not

Comment (taalexander) [reply qiskit/pulse/pulse_lib/samplers/decorators.py]: The Qiskit convention thus far seems to be to use relative imports, but it does not seem there is standardization throughout. Relative imports are not explicitly discouraged anymore in PEP 328. If they had been used originally in this module the line wouldn't have had to change when moving the module. Thoughts @ajavadia, @kdk?

Comment (taalexander) [reply qiskit/pulse/pulse_lib/samplers/decorators.py]: I should add, I'm ok either way.

Comment (lcapelluto) [reply qiskit/pulse/pulse_lib/samplers/decorators.py]: Ik qiskit usually uses relative imports; I guess I got excited that there was a file that had absolute imports 😁 I'm also alright either way

**Comment by taalexander** [comment]
I think this should be ready now.

**Pull request merged**